### PR TITLE
 sumologicexporter: refactor sender

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

Refactor sender in order to prepare for adding OTLP formats

Reduces size of #32315

Reopens #32592 due to revert due to test issues

**Link to tracking Issue:** #31479

**Testing:** 

Minimal changes to unit tests to ensure that everything works like before

**Documentation:**

N/A

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
Forget to create as draft 😓

**Comment by sumo-drosiek** [comment]
@mx-psi This is the same like #32592, but I added exception for goleak to `metadata.yaml` for opencensus view. I ses this is the common practice in the repo

File: exporter/sumologicexporter/metadata.yaml [hunk 1]
```diff
@@ -12,3 +12,8 @@ status:
 # TODO: Update the exporter to pass the tests
 tests:
   skip_lifecycle: true
+  goleak:
+    ignore:
+      top:
+        # See https://github.com/census-instrumentation/opencensus-go/issues/1191 for more information.
+        - "go.opencensus.io/stats/view.(*worker).start"
```
Comment (sumo-drosiek) [inline exporter/sumologicexporter/metadata.yaml:19]: Fix is here

**Comment by sumo-drosiek** [comment]
Not sure why correctness tests timed out 😕 
It seems to be unrelated to my changes

File: exporter/sumologicexporter/metadata.yaml (hunk 1 above)
Comment (mx-psi) [reply exporter/sumologicexporter/metadata.yaml:19]: @crobert-1 Looks good to me, but can you take a look at this exception before we merge this?

**Review by swiatekm — APPROVED** [review]

**Comment by crobert-1** [comment]
> Not sure why correctness tests timed out 😕 It seems to be unrelated to my changes

I didn't get a chance to see the failed action being referenced, but there's an open issue (https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27295) for a test that's timing out frequently, most likely unrelated to your changes 👍

File: exporter/sumologicexporter/metadata.yaml (hunk 1 above)
Comment (crobert-1) [reply exporter/sumologicexporter/metadata.yaml:19]: Yep, looks good. The opencensus package starts a goroutine upon init that has no way to be shutdown, so just importing the package causes a leak.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**